### PR TITLE
JupyterLab 3 update

### PR DESCRIPTION
**Pull request by martinRenou**

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -64,21 +64,21 @@
...
-    cmdclass        = cmdclass,
-    packages        = find_packages(),
-    author          = 'Mariana Meireles',
-    author_email    = 'mariana.meireles@quantstack.net',
-    url             = 'https://github.com/Quantstack/ipycytoscape',
-    license         = 'BSD',
-    platforms       = "Linux, Mac OS X, Windows",
-    keywords        = ['Jupyter', 'Widgets', 'IPython'],
-    classifiers     = [
+    name=name,
```
Comment (martinRenou) [inline setup.py:67]: This is for PEP8 compliance

**Comment by marimeireles** [comment]
@martinRenou just to know the state of it. Why can't we merge it?
It's not ready yet? Or external reasons?

**Comment by martinRenou** [comment]
We shouldn't merge this before JupyterLab 3 is released unless you want to make a release candidate that supports Jupyterlab 3? But that's a bit of extra work that IMO is not worth the pain, especially the number of widgets libraries that we have out there...

**Comment by marimeireles** [comment]
Noo
It's good, just wanted to know what's up. Thanks :)

**Comment by martinRenou** [comment]
As soon as JupyterLab 3 is released I'll go over the JupyterLab 3 PRs I seeded everywhere and finish them :)

**Comment by ianhi** [comment]
> I seeded everywhere and finish them :)

And then gaze triumphant over your forest of updated widgets. Overall heroic effort by you :)

**Comment by ianhi** [comment]
unmarked as draft because I wasn't able to push to this, but I think that something is just broken in my setup.

**Comment by martinRenou** [comment]
> Overall heroic effort by you

Heroic or foolish :P

I'll try to find an hour to update this one during the week. Maybe tonight.

File: src/plugin.ts [hunk 2]
```diff
@@ -36,8 +38,6 @@ function activateWidgetExtension(
   registry.registerWidget({
     name: MODULE_NAME,
     version: MODULE_VERSION,
-
-    exports: async () =>
-      await import(/* webpackChunkName: "jupyter-cytoscape" */ './index'),
```
Comment (ianhi) [inline src/plugin.ts:41]: Is this necessary to remove for jlab3? 

I don't really understand what this does but @bollwyvl added it intentionally so I assume it is a good thing.

Comment (martinRenou) [reply src/plugin.ts:41]: I recall it was for lazily loading the extension? But I understand this should not be needed with JupyterLab 3, I can be wrong though, I can add it back

**Comment by bollwyvl** [comment]
I recommend always lazy loading... Otherwise it blocks the moon animation,
instead of first paint of a widget.

File: src/plugin.ts (hunk 2 above)
Comment (ianhi) [reply src/plugin.ts:41]: I tried adding it back and got all kinds of fun errors when trying to create a wheel
```
Compilation finished

[
  {
    moduleIdentifier: '/home/ian/Documents/jupyter/ipycytoscape/lib/plugin.js',
    moduleName: './lib/plugin.js',
    loc: '35:83-144',
    message: `It's not allowed to load an initial chunk on demand. The chunk name "jupyter-cytoscape" is already used by an entrypoint.`,
    moduleId: 4480,
    moduleTrace: [ [Object] ],
    details: undefined,
    stack: `AsyncDependencyToInitialChunkError: It's not allowed to load an initial chunk on demand. The chunk name "jupyter-cytoscape" is already used by an entrypoint.\n` +
      '    at iteratorBlock (/home/ian/Documents/jupyter/ipycytoscape/node_modules/@jupyterlab/builder/node_modules/webpack/lib/buildChunkGraph.js:426:8)\n' +
      '    at processBlock (/home/ian/Documents/jupyter/ipycytoscape/node_modules/@jupyterlab/builder/node_modules/webpack/lib/buildChunkGraph.js:549:4)\n' +
      '    at processQueue (/home/ian/Documents/jupyter/ipycytoscape/node_modules/@jupyterlab/builder/node_modules/webpack/lib/buildChunkGraph.js:652:6)\n' +
      '    at visitModules (/home/ian/Documents/jupyter/ipycytoscape/node_modules/@jupyterlab/builder/node_modules/webpack/lib/buildChunkGraph.js:1111:3)\n' +
      '    at buildChunkGraph (/home/ian/Documents/jupyter/ipycytoscape/node_modules/@jupyterlab/builder/node_modules/webpack/lib/buildChunkGraph.js:1275:2)\n' +
      '    at Compilation.seal (/home/ian/Documents/jupyter/ipycytoscape/node_modules/@jupyterlab/builder/node_modules/webpack/lib/Compilation.js:2041:3)\n' +
      '    at /home/ian/Documents/jupyter/ipycytoscape/node_modules/@jupyterlab/builder/node_modules/webpack/lib/Compiler.js:1050:20\n' +
      '    at /home/ian/Documents/jupyter/ipycytoscape/node_modules/@jupyterlab/builder/node_modules/webpack/lib/Compilation.js:1877:4\n' +
      '    at _next4 (eval at create (/home/ian/Documents/jupyter/ipycytoscape/node_modules/@jupyterlab/builder/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:25:1)\n' +
      '    at eval (eval at create (/home/ian/Documents/jupyter/ipycytoscape/node_modules/@jupyterlab/builder/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:70:1)',
    compilerPath: undefined
  }
]
An error occured.
subprocess.CalledProcessError: Command '['node', '/home/ian/Documents/jupyter/ipycytoscape/node_modules/@jupyterlab/builder/lib/build-labextension.js', '--core-path', '/home/ian/anaconda3/envs/cytoscape/lib/python3.8/site-packages/jupyterlab/staging', '/home/ian/Documents/jupyter/ipycytoscape']' returned non-zero exit status 2.
See the log file for details:  /tmp/jupyterlab-debug-6gtf5vsu.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! jupyter-cytoscape@1.1.0 build:labextension: `npm run clean:labextension && jupyter labextension build .`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the jupyter-cytoscape@1.1.0 build:labextension script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/ian/.npm/_logs/2021-01-05T02_24_36_750Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! jupyter-cytoscape@1.1.0 build:all: `npm run build:labextension && npm run build:nbextension`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the jupyter-cytoscape@1.1.0 build:all script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/ian/.npm/_logs/2021-01-05T02_24_36_775Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! jupyter-cytoscape@1.1.0 build: `npm run build:lib && npm run build:all`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the jupyter-cytoscape@1.1.0 build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/ian/.npm/_logs/2021-01-05T02_24_36_795Z-debug.log
Traceback (most recent call last):
  File "setup.py", line 121, in <module>
    setup(**setup_args)
  File "/home/ian/anaconda3/envs/cytoscape/lib/python3.8/site-packages/setuptools/__init__.py", line 161, in setup
    return distutils.core.setup(**attrs)
  File "/home/ian/anaconda3/envs/cytoscape/lib/python3.8/distutils/core.py", line 148, in setup
    dist.run_commands()
  File "/home/ian/anaconda3/envs/cytoscape/lib/python3.8/distutils/dist.py", line 966, in run_commands
    self.run_command(cmd)
  File "/home/ian/anaconda3/envs/cytoscape/lib/python3.8/distutils/dist.py", line 985, in run_command
    cmd_obj.run()
  File "/home/ian/anaconda3/envs/cytoscape/lib/python3.8/site-packages/jupyter_packaging/setupbase.py", line 502, in run
    [self.run_command(cmd) for cmd in cmds]
  File "/home/ian/anaconda3/envs/cytoscape/lib/python3.8/site-packages/jupyter_packaging/setupbase.py", line 502, in <listcomp>
    [self.run_command(cmd) for cmd in cmds]
  File "/home/ian/anaconda3/envs/cytoscape/lib/python3.8/distutils/cmd.py", line 313, in run_command
    self.distribution.run_command(command)
  File "/home/ian/anaconda3/envs/cytoscape/lib/python3.8/distutils/dist.py", line 985, in run_command
    cmd_obj.run()
  File "/home/ian/anaconda3/envs/cytoscape/lib/python3.8/site-packages/jupyter_packaging/setupbase.py", line 274, in run
    c.run()
  File "/home/ian/anaconda3/envs/cytoscape/lib/python3.8/site-packages/jupyter_packaging/setupbase.py", line 380, in run
    run(npm_cmd + ['run', build_cmd], cwd=node_package)
  File "/home/ian/anaconda3/envs/cytoscape/lib/python3.8/site-packages/jupyter_packaging/setupbase.py", line 225, in run
    return subprocess.check_call(cmd, **kwargs)
  File "/home/ian/anaconda3/envs/cytoscape/lib/python3.8/subprocess.py", line 364, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['/home/ian/anaconda3/envs/cytoscape/bin/npm', 'run', 'build']' returned non-zero exit status 1.
```

Comment (ianhi) [reply src/plugin.ts:41]: Ideally we'd fix these errors and return to lazy loading, but for now I don't really konw how to fight that. So I'll open an issue about restoring it and we can always make a point release to add it back.

**Comment by ianhi** [comment]
Thanks @martinRenou !!

**Comment by bollwyvl** [comment]
>  The chunk name "jupyter-cytoscape"

just needs a different name. Will PR.

**Pull request merged**